### DISCUSSION TITLE
npm registry option

### DISCUSSION THREAD

**Discussion by BenSawary** [bikecoders/ngx-deploy-npm#388] [Ideas]
Hi,
I'm working with a production npm repository and a local Verdaccio repository. To avoid publishing to the wrong repository, it would be nice if there were an invocation parameter to select the repository (--registry xyz). Currently we have a [`publishConfig`](https://docs.npmjs.com/cli/v8/configuring-npm/package-json#publishconfig) option in our package.json that sometimes results in publishing to the wrong repository if someone isn't careful.

Kind regards,
ben

**Comment by dianjuar** [comment]
Interesting.
Can you tell me more about your use case on this thread 🧵 ⬇️

**Reply by dianjuar** [reply]
Let me do some questions to understand better:

> Currently we have a [publishConfig](https://docs.npmjs.com/cli/v8/configuring-npm/package-json#publishconfig) option in our package.json that sometimes results in publishing to the wrong repository if someone isn't careful.

Shouldn't [`publishConfig`](https://docs.npmjs.com/cli/v8/configuring-npm/package-json#publishconfig) option on the `package.json` file wouldn't be enough to prevent someone publishes it on the wrong repo?

Maybe the [`private`](https://docs.npmjs.com/cli/v8/configuring-npm/package-json#private) could help with the prevention.

---

> if someone isn't careful.

What steps do I need to follow to publish the package on the correct repo, and how can I mess it up?

---

I'm not super familiar with private npm repositories. I have used them in the past but as a consumer (to download packages).
I want to know more about your use case to understand what could be the best approach.

**Comment by dianjuar** [comment]
How I see it is to add an option as you suggested but on the generator `install` (AKA `ng-add`) that given a registry would set on the `package.json` file:
- The [`publishConfig`](https://docs.npmjs.com/cli/v8/configuring-npm/package-json#publishconfig) option
- The option [`private`](https://docs.npmjs.com/cli/v8/configuring-npm/package-json#private) to `true` to and I quote
  - > to prevent accidental publication of private repositories

**Comment by dianjuar** [comment]
Feature implemented by @dmitry-stepanenko on #483
